<?xml version='1.0'?>
<launch>
  
  <arg name="robot_ns"/>

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find prius_control)/config/prius_control.yaml" command="load"/>

  <!-- load the controllers -->
  <node name="controller_manager" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/$(arg robot_ns)" args="rear_left_wheel_velocity_controller rear_right_wheel_velocity_controller
                                            front_left_wheel_velocity_controller front_right_wheel_velocity_controller
                                            front_left_steer_position_controller front_right_steer_position_controller
                                            joint_state_controller"/>

<!-- robot_state_publisher has already taken care of this -->
  <!-- TF Setting 
  <include file="$(find od_control)/launch/includes/tf2.xml"/>
 --> 

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
   <remap from="/joint_states" to="/$(arg robot_ns)/joint_states" /> 
  </node> 
  
  <!-- driving node -->
  <node pkg="prius_control" type="prius_driving_commands.py" name="servo_commands" output="screen">
  </node>

  <!-- Allow for Gazebo to broadcast odom, publish tf from /map -> /odom --> 
  <node pkg="prius_gazebo" name="gazebo_odometry_node" type="gazebo_odometry_notransform.py">
  </node>

</launch>
